core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -174,15 +162,41 @@ oppia.directive('audioControls', [
               AudioPlayerService.getCurrentAudioControlsDirectiveId();
           };
 
+          var onFinishedLoadingAudio = function(loadedFilename) {
+            if (audioRequestIsPending &&
+                loadedFilename === 
+                  AudioPreloaderService
+                    .getMostRecentlyRequestedAudioOnCurrentCard()) {
+              audioRequestIsPending = false;
+              AudioPlayerService.load(loadedFilename, directiveId)
+                .then(function() {
+                  AudioPlayerService.play();
+                });
+            }
+            $scope.audioIsLoading = false;
```

Comment:
What is the difference between $scope.audioIsLoading and audioRequestIsPending?

If the former is for the display in the directive, then perhaps consider the name $scope.audioLoadingIndicatorIsShown or similar.

Reply:
Done